### ISSUE TITLE
Gradient Support does not work correctly on Aggregate Graphs

### ISSUE THREAD

**Issue by ISekan** [Cacti/cacti#5267]
Cacti 1.2.24
It works properly only for first item.
Is this a bug or a feature?
![image](https://user-images.githubusercontent.com/17400240/224232440-b698b6dd-085c-4339-a342-20296161b900.png)
![image](https://user-images.githubusercontent.com/17400240/224232720-97378581-abaf-489a-bc1b-97fd43a9f055.png)

**Comment by TheWitness** [comment]
I looked at making it work with AREA:STACK and it did not look good.  So, I reverted it.  It's an open issue though.

**Comment by TheWitness** [comment]
It's more of a technical issue as the method that the contributor used basically renders several quadrants over one another with varying opacity.  Each area renders from the bottom of the Y-Axis upto the path.  So, the final gradient becomes all washed out, and it also does not work well with AREA's on top of AREA's due to the fact the design does not allow for Alpha Transparency.  It's all technical, might be able to be fixed, but it was beyond the 1.2.x scope.  Maybe we can do something in 1.3.x.

**Comment by thurban** [comment]
To be clear for going forward:
- Gradient support will be per graph template instead of global
- Support for a light/dark mode (color going to a lightened or darkened version)
- Changes will be in 1.3.x only

**Comment by ISekan** [comment]
[thurban](https://github.com/thurban) of course

![image](https://user-images.githubusercontent.com/17400240/229480697-e017f3b2-5428-4ec1-a3d1-dd096b422e3a.png)

![image](https://user-images.githubusercontent.com/17400240/229481715-eda127d0-7eef-4c3b-bd74-5aef1c68ba78.png)

**Comment by thurban** [comment]
Yeah. Unfortunaltey this seems to be a rrdtool bug.  If you disable "slopemode" on the graphs, it looks better, but rrdtool seems to slightly overlap the different areas, resulting in these lines.  It doesn't show when no transparency is being used as it is a solid color then.

I'm looking into finding a fix for that, but it's going to take a while.

**Comment by thurban** [comment]
@ISekan It looks like the version breaks the "Thumbnails". Can you confirm ?

**Comment by thurban** [comment]
@ISekan you do not happen to have the possibility to compile rrdtool yourself ?

**Comment by thurban** [comment]
Opened a bug report with rrdtool:
https://github.com/oetiker/rrdtool-1.x/issues/1219

**Comment by ISekan** [comment]
> you do not happen to have the possibility to compile rrdtool yourself ?
Unfortunately not
I have ver. RRD 1.7.1 on the cacti server

**Comment by ISekan** [comment]
> Seems like this only occurrs under very specific conditions ( very short time range )
poller time is 1min
If time preset is set 12h - all's ok, After switch to 1 day or more it is look no so good as you can see below
![image](https://user-images.githubusercontent.com/17400240/230305015-486d5fba-514e-4fb6-a1bb-6f0c99cecb17.png)
![image](https://user-images.githubusercontent.com/17400240/230305069-c05aaa3a-4361-4ec3-8483-945677dcd2c5.png)

**Comment by thurban** [comment]
Hi,

Not sure, but I don't see any gradient there, do I miss something ?

**Comment by TheWitness** [comment]
@thurban,

Read this.  Are you a taker on the change?

```
AREA:value[#color[#color2]][:[legend][:STACK][:skipscale][:gradheight=y]

See LINE, however the area between the x-axis and the line will be filled.

If color2 is specified, the area will be filled with a gradient.

The gradheight parameter can create three different behaviors. 
If gradheight > 0, then the gradient is a fixed height, 
starting at the line going down. 

If gradheight < 0, then the gradient starts at a fixed height above the x-axis, 
going down to the x-axis. If height == 0, then the gradient goes from the 
line to x-axis.

The default value for gradheight is 50.
```

**Comment by TheWitness** [comment]
That did not take that long.  Requires QA...  I might change the wording a bit.  Should we dump the previous Gradient support?

![image](https://github.com/user-attachments/assets/da7c4dc8-5181-4efb-985c-5d5a95a295d7)


![image](https://github.com/user-attachments/assets/7c27558d-17a1-43b5-a5b1-8706be9fb8eb)

**Comment by TheWitness** [comment]
Fixed in #5212.  Updating the ChangeLog shortly.

**Comment by TheWitness** [comment]
Example from a Graph Template with 4 CPU's using Totaling.

![image](https://github.com/user-attachments/assets/e7ed6648-e2ae-4d12-886a-72155c5d5c9a)
